src/Native/jitinterface/corinfoexception.cpp

```diff
@@ -0,0 +1,50 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#include "corinfoexception.h"
+
+class CorInfoExceptionList
+{
+public:
+    CorInfoExceptionList(const WCHAR* message, int messageLength, CorInfoExceptionList *pNext)
+        : _ex(message, messageLength),
+        _pNext(pNext)
+    {
+    }
+    ~CorInfoExceptionList()
+    {
+        delete _pNext;
+    }
+    CorInfoException* GetException()
+    {
+        return &_ex;
+    }
+
+private:
+    CorInfoException _ex;
+    CorInfoExceptionList* _pNext;
+};
+
+static CorInfoExceptionList *pExceptionList = nullptr;
```

Comment:
Fixed by tracking the exceptions in a List in managed code. Insertion into the List is protected by lock to be multi-threading safe.